### ISSUE TITLE
[bug] install with lockfile does not respect profile settings (from within the lockfile) when config_options are used

### ISSUE THREAD

**Issue by a4z** [conan-io/conan#8091]
### Environment Details (include every applicable attribute)
  * Operating: OSX Catalina
  * Compiler+version Apple Clang 12 
  * Conan version: Conan version 1.31.3
  * Python version: Python 3.8.5

### Steps to reproduce (Include if Applicable)

install via lockfile does not respect profile settings (from within the lockfile) when config_options are used

* create a new conan project

* add libcurl as requirement

* have profiles that set the option to build libcurl with openssl, also on OXS. iOS 

* generate a lockfile (via conan install)

* use the lockfile to install dependencies 

```
ERROR: libcurl/7.71.1: YourApp/0.0.1 tried to change libcurl/7.71.1 option with_ssl to openssl
but it was already defined as darwinssl
```

but thus is not true, or a problematic truth,
since the lockfile contains settings for libcurl and no user action has been taken to make curl use darwinssl
darwinssl is a default that is set from within the recipe

```
Using lockfile: '/Users/a4z/lockfiletest/build/gen1/Darwin_nsdk-ios-x86_64_conan.lock'
Configuration (profile_host):
[settings]
arch=x86_64
build_type=Release
compiler=apple-clang
compiler.cppstd=17
compiler.libcxx=libc++
compiler.version=12.0
os=iOS
os.version=11.0
[options]
ios-cmake:toolchain_target=SIMULATOR64
libcurl:darwin_ssl=False
libcurl:with_ssl=openssl
lua:compile_as_cpp=True
ms-gsl:on_contract_violation=throw
nlohmann_json:multiple_headers=True
paho-mqtt-c:asynchronous=True
paho-mqtt-c:shared=False
paho-mqtt-c:ssl=True
paho-mqtt-cpp:ssl=True
[build_requires]
*: ios-cmake/3.1.2, ios-cmake/3.1.2
[env]
CONAN_CMAKE_GENERATOR=Xcode
libcurl:CFLAGS=-fembed-bitcode
libcurl:CPPFLAGS=-fembed-bitcode
openssl:CFLAGS=-fembed-bitcode
Configuration (profile_build):
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=apple-clang
compiler.libcxx=libc++
compiler.version=12.0
os=Macos
os_build=Macos
[options]
[build_requires]
[env]
```

the libcurl recipe has config_options
```
    def config_options(self):
        if self.settings.os == "Windows":
            del self.options.fPIC
        if not self._has_zstd_option:
            del self.options.with_zstd
        # Default options
        self.options.with_ssl = "darwinssl" if tools.is_apple_os(self.settings.os) else "openssl"
```

it is this line that produces the wrong error
```
        self.options.with_ssl = "darwinssl" if tools.is_apple_os(self.settings.os) else "openssl"
```

removing this, everything works as it should

see also this thread 
https://cpplang.slack.com/archives/C41CWV9HA/p1605879913068300

**Comment by memsharded** [comment]
This should be fixed by https://github.com/conan-io/conan/pull/7993, which is planned for next 1.32, apparently same issue as https://github.com/conan-io/conan/issues/7991

**Comment by a4z** [comment]
this seems to be fixed in 1.32, at least my first tests do not trigger that problem anymore after adding the line back to the recipe.
Well and fast done, thanks ! 

so I close this issue now
